src/Nethermind/Nethermind.Core/IKeyValueStore.cs

```diff
@@ -19,6 +22,16 @@ public interface IReadOnlyKeyValueStore
         byte[]? this[ReadOnlySpan<byte> key] => Get(key);
 
         byte[]? Get(ReadOnlySpan<byte> key, ReadFlags flags = ReadFlags.None);
+
+        /// <summary>
+        /// Return span. Must call `DangerousReleaseMemory` or there can be some leak.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>Can return null or empty Span on missing key</returns>
+        Span<byte> GetSpan(ReadOnlySpan<byte> key, ReadFlags flags = ReadFlags.None) => Get(key, flags);
+
+        bool KeyExists(ReadOnlySpan<byte> key) => GetSpan(key).IsNull();
+        void DangerousReleaseMemory(in Span<byte> span) { }
```

Comment:
What do you think - remove this method and in `GetSpan` return ref struct that also has Dispose method and use dispose pattern?

Reply:
I'd recommend that. See `GetOwnedMemory`. It is however, marginally less efficient. Can also do `GetSpan(ReadOnlySpan<byte> key, Func<Span<byte>,TReturn> ret)`, which is the same interface as rocksdb.